web/client/components/data/featuregrid/formatters/index.js

```diff
@@ -26,7 +26,7 @@ const dateTimeFormatter = ({value, format, type}) => {
             ? moment.utc(value).format(format)
             : type === 'time'
                 ? moment(`${DEFAULT_DATE_PART}T${value}`).utc().format(format) // time format append default date part
-                : moment(value).format(format) // date or date-time formats
+                : moment(value, format).format(format) // date or date-time formats
```

Comment:
The input **format** should not be the same as the output `format`. This function is taking a raw date/date-time/time _value_ from feature and convert it to configured format for visualization so we expect:

moment(_value_, **'YYYY-MM-DD[Z]'**).format(`format`)

We should apply the input **format** when the type is only date and the _value_ ends with **Z**, in general is a good practice to apply the input format when known as suggested [here](https://github.com/geosolutions-it/MapStore2/issues/9275#issuecomment-1655403718).
The current implementation is not working if you try to replace the [localConfig](https://github.com/geosolutions-it/MapStore2/blob/1f30c78dea9212cdcc9ba6890cc45c3ff643e0d9/web/client/configs/localConfig.json#L438) configuration of [FeatureEditor](https://mapstore.geosolutionsgroup.com/mapstore/docs/api/plugins#plugins.FeatureEditor) with custom formats:

```diff
- "FeatureEditor",
+ {
+  "name": "FeatureEditor",
+  "cfg": {
+    "dateFormats": {
+      "date-time": "MM DD YYYY - HH:mm:ss",
+      "date": "MM DD YYYY",
+      "time": "HH:mm:ss"
+    }
+  }
+ },
```

Reply:
I check momentjs documentation for this point and I saw something I think it may be helpful. 
If I don't know the exact input format, I can pass all the available date formats in an array (list) and it will match the correct one from the list.
check this image: 
![image](https://github.com/geosolutions-it/MapStore2/assets/58145645/f1694ead-970b-4206-8125-e8ffd02bd310)

ref: https://momentjs.com/docs/#/parsing/string-formats/